pipeline/pipeline_tasks/api_calls/geocode_control.py

```diff
@@ -0,0 +1,240 @@
+def check_credentials(args):
+    try:
+        if args.geocode_key is None and os.environ['GOOGLE_MAPS_API'] is None:
+            return False
+    except:
+        return False
+    return True
+
+
+# Set the parameters for Google Maps. The maximum number of queries per day is 25,000.
+# We will store the data no more than 21 days.
+# The user can specify values on the command line, but only values smaller than the max are honored.
+# Returns params in a dictionary.
+def get_params(max_records, older_than):
+    params = { 'max_records': 2500, 'max_days_to_store': 30 }
+    if max_records > 0:
+        params['max_records'] = min(params['max_records'], max_records)
+    if older_than > 0:
+        params['max_days_to_store'] = min(params['max_days_to_store'], older_than)
+    return params
+
+
+# This actually gets the records to update, calls the API function and writes back to the database.
+# Returns the number of records updated, or None if a serious error occurred.
+# Can return 0 if nothing found to update.
+def update_records(args, api_params, db_params):
+    # Create a DB manager object and a pandas dataframe with just the set of records to be updated.
+    # Then call the Google Maps API for each entry in the dataframe and update if it works.
+    max_records = api_params['max_records']
+    max_days_to_store = api_params['max_days_to_store']
+    db_url = db_params['db_url']
+    dbm = DBManager(db_url=db_url)
+
+    sfdo_update = get_records(dbm, max_records, max_days_to_store)
+    if sfdo_update is None:
+        return None
+    elif len(sfdo_update) < 1:
+        return 0
+
+    update_count = update_geocode(args, sfdo_update)
+    if update_count is not None:
+        sfdo_orig = get_all_records(dbm)
+        if sfdo_orig is None:
+            return None
+        elif len(sfdo_orig) < 1:
+            return 0
+        sfdo_update.drop('borr_name', axis=1, inplace=True)
+        sfdo_update.drop('borr_street', axis=1, inplace=True)
+        sfdo_update.drop('borr_city', axis=1, inplace=True)
+        sfdo_update.drop('borr_state', axis=1, inplace=True)
+        sfdo_update.drop('borr_zip', axis=1, inplace=True)
+        sfdo_orig = sfdo_orig.set_index('sba_sfdo_id')
+        sfdo_update = sfdo_update.set_index('sba_sfdo_id')
+        sfdo_orig.update(sfdo_update)
+        print('......Save Geocode data updates')
+        dbm.write_df_table(sfdo_orig, table_name='sba_sfdo_api_calls', schema='stg_analytics', index=True)
+
+    return update_count
+
+
+# This method will erase all the timestamps for the API.  The effect
+# is that the batch update process will then need to update every
+# record.
+def reset_timestamp(db_params):
+    print('......Clear all Google Geocode timestamps')
+    db_url = db_params['db_url']
+    dbm = DBManager(db_url=db_url)
+    sfdo_orig = get_all_records(dbm)
+    sfdo_orig['geocode_timestamp'] = pd.to_datetime('None', errors='coerce')
```

Comment:
Should this be `pd.to_datetime(None, errors='coerce')`

Reply:
Note to self, problem is that this is resulting in `geocode_timestamp` to be written as a text which is causing problems.

Reply:
to_datetime(None) or to_datetime('None') seems to change the column type to text.